rviz_simulator/src/simulate.cpp

```diff
@@ -155,57 +218,75 @@ int main(int argc, char** argv)
 
   g_interactive_marker_server.reset(new interactive_markers::InteractiveMarkerServer("simulate", "", false));
 
-  ros::Duration(0.1).sleep();
+  ros::Duration(1.0).sleep();
 
   // loading parameters
+  // getting the world_frame_id
   std::string world_frame_id;
-  n.getParam("/world_frame_id", world_frame_id);
+  if (!n.getParam("/world_frame_id", world_frame_id))
+  {
+    world_frame_id = "ROSWorld";
+    ROS_WARN_STREAM("\"world_frame_id\" param was not specified. Using \"" << world_frame_id << "\" as the "
+                                                                                                "world_frame_id.");
+  }
 
   /// setting target marker color and scale
-  std_msgs::ColorRGBA grey = loadColor(n, "grey");
-  std_msgs::ColorRGBA blue = loadColor(n, "blue");
-  double target_scale;
-  n.getParam("target_scale", target_scale);
+  double target_scale = 0.1;
```

Comment:
I made the param in the `initialize_simulator.yaml` file a sequence of length 3 (length, width, depth), example:
`target_size: [0.1, 0.1, 0.01]`
which is read into a `std::vector<double>` from the parameter server in the `simulate.cpp` file (since rosparam cpp can automatically read vectors with no additional code)

The depth is theoretically 0 since the target is a plane, so I gave it a depth of 1cm and set the target class to accept the vector and initialize the visualization marker dimensions accordingly.